tests/http/httpClient/Middleware/AuthorizationHandlerTests.cs

```diff
@@ -0,0 +1,117 @@
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using Microsoft.Kiota.Abstractions.Authentication;
+using Microsoft.Kiota.Http.HttpClientLibrary.Middleware;
+using Microsoft.Kiota.Http.HttpClientLibrary.Tests.Mocks;
+using Moq;
+using Xunit;
+
+namespace Microsoft.Kiota.Http.HttpClientLibrary.Tests.Middleware
+{
+    public class AuthorizationHandlerTests : IDisposable
+    {
+        private readonly MockRedirectHandler _testHttpMessageHandler;
```

Comment:
Since this is shared in the class, you may run into problems when the multiple tests use it as each instance is currently meant to only send one request. 

https://github.com/microsoft/kiota-dotnet/blob/75f6bd3ae9d4bff6ad0c9a1f38cbf5be7c7289d6/tests/http/httpClient/Mocks/MockRedirectHandler.cs#L18

You should probably use the `TestHttpMessageHandler` type instead

Reply:
`MockRedirectHandler` allowed me to chain multiple requests easily since the Auth handler retries only once if a CAE challenge is returned by the first request: [here](https://github.com/microsoft/kiota-dotnet/pull/430/files#diff-d10ea300ecbd16c34c6980e04b9b192f660a99f284b061754539ce34c41d3448R122)

Reply:
I would suggest removing this as a class variable and initializing a new varible in the tests. Since the tests are not guaranteed to be executed sequentially depending on test context, you may run into flaky tests...

```cs
var testHttpMessageHandler = new MockRedirectHandler();
```

Reply:
This would mean initialising a few more variables currently in the test class constructor within every test.
Curious why the tests might get flaky?
I thought each test will invoke the constructor & the class variables will be re-initialised, so each test uses a different instance of the handler. Similar approach is being used in the Redirect hander [tests](https://github.com/microsoft/kiota-dotnet/blob/main/tests/http/httpClient/Middleware/RedirectHandlerTests.cs)

Reply:
Yeah. Your right. Resolving this for now.
https://xunit.net/docs/shared-context